src/main/java/com/automatedtest/sample/pages/SettingPage.java

```diff
@@ -64,16 +66,9 @@ public void saveSettingPage() {
         this.wait.forLoading();
     }
 
-    //TODO: support generic dynamic element
-    public WebElement getDivRadioSpokenByValue(String value) {
-        String sxpath = "//span[text()='" + value + "']/parent::div";
-        WebElement div = this.driver.findElement(By.xpath(sxpath));
-        return div;
-    }
-
     public void selectRadioSpoken(String value) {
         // get radio have text equal : value
-        WebElement div = getDivRadioSpokenByValue(value);
+        WebElement div = this.driver.findElement(By.xpath(link.format(value)));
```

Comment:
you should support the `findElement` and `findElements` inside of the `Link` class.
it should take only the `value` as parameter, for the driver, we could get it directly from `DriverUtils.getDriver` (after the PR about run test in parallel is merged, it should be available)